Tests/RegexBuilderTests/AlgorithmsTests.swift

```diff
@@ -104,4 +104,38 @@ class RegexConsumerTests: XCTestCase {
        result: "9+16, 3, 10, 99+1")
     )
   }
+
+  func testSwitches() {
+    switch "abcde" {
+    case Regex {
+      "a"
+      ZeroOrMore(.any)
+      "f"
+    }:
+      XCTFail()
+
+    case "abc":
```

Comment:
Is there a way to test that this is hitting the `Equatable` version instead of the `RegexComponent` version?

Reply:
I don't think so. Any suggestions?

Reply:
I don't have an idea that doesn't require making both sides generic, which I don't think we want to do. :/ We can check temporarily by printing / breaking in the operator function, but there's not really a way to test it properly.